src/Host/Monitor/Impl/MainWindow.xaml.cs

```diff
@@ -13,53 +16,52 @@ public partial class MainWindow : Window {
             InitializeComponent();
             Loaded += MainWindow_Loaded;
             _currentWindow = this;
-            Closing += MainWindow_Closing;
+            UseBrokerUserCheckBox.IsChecked = Properties.Settings.Default.UseDifferentBrokerUser;
         }
 
-        private async void MainWindow_Closing(object sender, System.ComponentModel.CancelEventArgs e) {
-            await BrokerManager.StopBrokerInstanceAsync();
+        private void MainWindow_Loaded(object sender, RoutedEventArgs e) {
+            StartUpAsync().DoNotWait();
         }
 
-        private async void MainWindow_Loaded(object sender, RoutedEventArgs e) {
-            if (Properties.Settings.Default.UseDifferentBrokerUser) {
-                if (await CredentialManager.IsBrokerUserCredentialSavedAsync()) {
-                    await BrokerManager.StartBrokerInstanceAsync();
+        public static async Task StartUpAsync() {
+            try {
+                if (Properties.Settings.Default.UseDifferentBrokerUser) {
+                    if (await CredentialManager.IsBrokerUserCredentialSavedAsync()) {
+                        await BrokerManager.CreateOrAttachToBrokerInstanceAsync();
+                    } else {
+                        await CredentialManager.GetAndSaveCredentialsFromUserAsync();
+                    }
                 } else {
-                    await CredentialManager.GetAndSaveCredentialsFromUserAsync();
+                    await BrokerManager.CreateOrAttachToBrokerInstanceAsync();
                 }
-            } else {
-                await BrokerManager.StartBrokerInstanceAsync();
+            } catch (Exception ex) {
```

Comment:
Can you add `when (!ex.IsCriticalException())` here? We don't want to swallow NRE and that kind of stuff.

Also, is there anything that will log the exception before this point? If not, we should log here, to diagnose issues.